types/txresponse.go

```diff
@@ -1,10 +1,57 @@
 package types
 
 import (
+	"encoding/json"
+
 	"github.com/cosmos/relayer/v2/relayer/provider"
 )
 
-type TxResponse struct {
+// TxResponse is the interface that handles different types of transaction responses
+type TxResponse interface {
+	GetTxHash() string
+	GetEventsBytes() []byte
+}
+
+var _ TxResponse = &BabylonTxResponse{}
+
+type BabylonTxResponse struct {
```

Comment:
```suggestion
type CosmosTxResponse struct {
```

This response is generic to all Cosmos SDK chains. For example, other Cosmos SDK chains that integrate with Babylon via a Go module could also use this response.

Reply:
I prefer to update the [CosmwasmTxResponse](https://github.com/babylonchain/finality-provider/blob/refactor/tx-response/types/txresponse.go#L36-L45) with the below definition:
```Go
type CosmosTxResponse struct {
	TxHash string
	Events []struct {
		Type       string
		Attributes []struct {
			Key   string
			Value string
			Index bool
		}
	}
}
```
It is more generic to all Cosmos SDK chains b/c it is part of the [sdk.TxRespons](https://github.com/cosmos/cosmos-sdk/blob/main/types/abci.pb.go#L34).

Reply:
`provider.RelayerEvent` can be converted to the event type in `sdk.TxResponse` right? Maybe we can do the conversion here?

Reply:
yes, we can convert it here.